WPThemeReview/Sniffs/Templates/RequiredFunctionSniff.php

```diff
@@ -0,0 +1,231 @@
+<?php
+/**
+ * WPThemeReview Coding Standard.
+ *
+ * @package WPTRT\WPThemeReview
+ * @link    https://github.com/WPTRT/WPThemeReview
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WPThemeReview\Sniffs\Templates;
+
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Util\Tokens;
+
+/**
+ * Ensures that body_class() is called if theme adds a <body> tag.
+ *
+ * @link  https://make.wordpress.org/themes/handbook/review/required/#templates
+ *
+ * @since 0.1.0
+ */
+class RequiredFunctionSniff implements Sniff {
+
+	/**
+	 * Sniff Settings
+	 *
+	 * @var array
+	 */
+	public $tagsConfig = array(
+		'body' => array(
+			'function'  => 'body_class',
+			'attribute' => 'class',
+		),
+		'html' => array(
+			'function'  => 'language_attributes',
+			'attribute' => 'lang',
+		),
+	);
+
+	/**
+	 * Supported Tokenizers
+	 *
+	 * Currently this sniff is only useful in PHP as the required
+	 * functions to call are done in PHP. In testing various
+	 * themes - some had inline comments including `<html>`, and
+	 * were tokenized as T_INLINE_HTML throwing some false positives.
+	 *
+	 * @var array
+	 */
+	public $supportedTokenizers = array( 'PHP' );
+
+	/**
+	 * Tag being searched.
+	 *
+	 * @var array
+	 */
+	protected $tag;
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return Tokens::$textStringTokens;
+	}
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @param \PHP_CodeSniffer\Files\File $phpcsFile The PHP_CodeSniffer file where the
+	 *                                               token was found.
+	 * @param int                         $stackPtr  The position of the current token
+	 *                                               in the stack.
+	 *
+	 * @return void
+	 */
+	public function process( File $phpcsFile, $stackPtr ) {
+
+		$tokens   = $phpcsFile->getTokens();
+		$content  = $this->clean_str( $tokens[ $stackPtr ]['content'] );
+		$filename = $phpcsFile->getFileName();
+
+		// Set to false if it is the first time this sniff is run on a file.
+		if ( ! isset( $this->tag[ $filename ] ) ) {
+			$this->tag[ $filename ] = false;
+		}
+
+		// Skip on empty.
+		if ( '' === $content ) {
```

Comment:
I don't think Yoda is forced on `WPThemeReview` ruleset 😄

Reply:
It is as WPCS still enforces it.